pkg/apis/batch/validation/validation.go

```diff
@@ -425,6 +548,26 @@ func validateJobStatus(status *batch.JobStatus, fldPath *field.Path) field.Error
 			}
 		}
 	}
+	if opts.RejectSuccessCriteriaMetForNonIndexedJob {
+		if IsConditionTrue(job.Status.Conditions, batch.JobSuccessCriteriaMet) {
```

Comment:
I'm not seeing the check this is a non-indexed job, please add that.

Reply:
We already check the mode here: https://github.com/kubernetes/kubernetes/blob/241dbdebaf0b4a0c0829e6ffc49ada94313c461a/pkg/registry/batch/job/strategy.go#L359

If so, should I check the mode here as well? Please let me know if I'm missing anything else.

Reply:
I think this should be only what triggers the check. But the check itself should be atomic.

Reply:
In the strategy the trigger should be a field change. This change also applies on creation. Even if the field is immutable later.

Reply:
> In the strategy the trigger should be a field change. This change also applies on creation. Even if the field is immutable later.

@atiratree I see. Let me check if my understanding is correct. We should move the `isIndexed` check from strategy to validation since we should have triggers only related to field changes in the strategy, right?

Reply:
- You should check the change between the `oldJob.Spec.CompletionMode` and `newJob.Spec.CompletionMode` in the strategy as a trigger. This will be only triggered on job creation as it is immutable.
- The validation should include the isIndexed check.

Reply:
Thank you for this confirmation.

Reply:
I'll try it.

Reply:
Done.

Reply:
I addressed Aldo's commets.